### ISSUE TITLE
Support list input of st2_packs

### ISSUE THREAD

**Issue by mickmcgrath13** [bitovi/github-actions-deploy-stackstorm#20]
Current `st2_packs` value is a comma separated string which might result in a long, hard to parse string like:
```
st2_packs: "st2=v1.0.7,circle_ci=v0.5.3,packagecloud=v1.0.1,aws=v1.2.1,sensu=v0.5.2,slack=v0.12.7,github=v1.0.0,https://github.com/StackStorm/GitHubWebhook,https://github.com/StackStorm/st2ci,https://github.com/StackStorm/st2cd,windows=v0.6.0,https://github.com/StackStorm/stackstorm-st2community"
```

Ideally, we could provide a list instead like:
```
st2_packs:
  - st2=v1.0.7
  - circle_ci=v0.5.3
  - packagecloud=v1.0.1
  - aws=v1.2.1
  - etc
```

**Comment by mickmcgrath13** [comment]
Relevant PR: https://github.com/bitovi/github-actions-deploy-stackstorm/pull/2#discussion_r1058991731

**Comment by LeoDiazL** [comment]
What about a checked in file or a github secret?
If the list get's too long the user might prefer to handle this in a file instead

**Comment by mickmcgrath13** [comment]
checked in file could work well, I think.  Don't think github secrets is the way to go because these packs are not sensitive and would value form clearly visible config

**Comment by armab** [comment]
If we want to control it via file, then just allowing to specify the user's Ansible vars would work perfectly and not only for packs, but all the StackStorm deployment configurations, see (https://github.com/bitovi/github-actions-deploy-stackstorm/issues/7#issuecomment-1370101010).

If we want to still control it via GH actions config, then it's more useful to have them as a list instead of comma-separated string:
https://github.com/bitovi/github-actions-deploy-stackstorm/blob/8b28995ef56abd428d5cbba247fbca9e1fb246d9/action.yaml#L59-L61

**Comment by PhillypHenning** [comment]
Looks like list / multiline support if coming as per https://github.com/actions/toolkit/issues/184
Though it's been in dev for roughly a year+ at this point so don't hold your breath for its conclusion

**Comment by armab** [comment]
Thanks for the research!
That's helpful, didn't know it's not natively supported by the GH Actions themselves.

So we leave it as is until proper support from Github.

**Comment by PhillypHenning** [comment]
So I currently have two solutions (as PRs) for this; 

### Provide a string input to the GHA that represents the path from the application root directory to a file that has a list of packs.

**example GHA input**
`dev/packs.txt`

**packs.txt**
```
- st2=v1.0.7
- circle_ci=v0.5.3
- packagecloud=v1.0.1
- aws=v1.2.1
- etc
```

### Provide a string input to the GHA that represents the path from the application root directory to the st2.conf file

**Comment by mickmcgrath13** [comment]
could this just be handled with the ansible variables st2.yaml?
https://github.com/bitovi/github-actions-deploy-stackstorm/pull/32

**Comment by armab** [comment]
We can close and forget about this issue when https://github.com/bitovi/github-actions-deploy-stackstorm/pull/32#pullrequestreview-1243007204 is merged and so packs install list could be overriden by the custom user-specified ansible vars file.
No need for another abstraction or dedicated file.

This way we allow 2 user flows:
1) simplified via GH Actions config (1st day - simple deployment)
2) advanced via dedicated Ansible vars (2nd day - complex configuration)

The 1st should be easy as possible and in this context the string of packs not just fine, but also fits perfectly.
The 2nd approach allows more detailed configuration.

We'll encourage users to start with 1st approach to get started and in case of more detailed configuration, continue with 2nd approach (including defaults overrides) for StackStorm deployment via Ansible vars.

Hope that story would make sense.
